src/pkg/notifier/formats/cdevents.go

```diff
@@ -0,0 +1,174 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package formats
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	cdevents "github.com/cdevents/sdk-go/pkg/api"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+	purl "github.com/package-url/packageurl-go"
+
+	"github.com/goharbor/harbor/src/controller/event"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/pkg/notifier/model"
+)
+
+var (
+	// cloudEventsFormatter is the global single formatter for CloudEvents.
+	cdEventsFormatter Formatter = &CDEventsFormatter{}
+)
+
+func init() {
+	registerFormats(CDEventsFormat, cdEventsFormatter)
+}
+
+const (
+	// CDEventsFormat is the type for CDEvents format.
+	CDEventsFormat = "CDEvents"
+)
+
+// cdEventFromHarborType
+func cdEventFromHarborType(he *model.HookEvent) (cdevents.CDEvent, error) {
+	// Only TopicPullArtifact is supported for now.
+	// Other event types are not striclty a failure (they are missing by design),
+	// but we still report them upstream as unknown types
+	if he.EventType != event.TopicPushArtifact {
```

Comment:
Yep, I agreed that other event types are not a failure case, user can subscribe other event types from UI but can only receive PUSH events, IMO the better solution is that extend the capability to support limit the event types.